src/panels/config/zone/dialog-home-zone-detail.ts

```diff
@@ -63,7 +64,7 @@ class DialogHomeZoneDetail extends LitElement {
         escapeKeyAction
         .heading=${createCloseHeading(
           this.hass,
-          this.hass!.localize("ui.panel.config.zone.edit_home")
+          `${this.hass!.localize("ui.common.edit")} ${this._data.name}`
```

Comment:
You should pass the name as context, so the word order can be changed for other languages. Same for the other instances
```suggestion
          this.hass!.localize("ui.common.edit", {name: this._data.name})
```